examples/email-recovery/src/pages/index.tsx

```diff
@@ -0,0 +1,243 @@
+const es256 = -7;
+const rs256 = -257;
+
+// This constant designates the type of credential we want to create.
+// The enum only supports one value, "public-key"
+// https://www.w3.org/TR/webauthn-2/#enumdef-publickeycredentialtype
+const publicKey = "public-key";
+
+const generateRandomBuffer = (): ArrayBuffer => {
+  const arr = new Uint8Array(32);
+  crypto.getRandomValues(arr);
+  return arr.buffer;
+};
+
+const base64UrlEncode = (challenge: ArrayBuffer): string => {
+  return Buffer.from(challenge)
+    .toString("base64")
+    .replace(/\+/g, "-")
+    .replace(/\//g, "_")
+    .replace(/=/g, "");
+};
+
+export default function RecoveryPage() {
+  const [initRecoveryResponse, setInitRecoveryResponse] =
+    useState<InitRecoveryResponse | null>(null);
+  const [iframeStamper, setIframeStamper] = useState<IframeStamper | null>(
+    null
+  );
+  const {
+    register: initRecoveryFormRegister,
+    handleSubmit: initRecoveryFormSubmit,
+  } = useForm<InitRecoveryFormData>();
+  const {
+    register: recoverUserFormRegister,
+    handleSubmit: recoverUserFormSubmit,
+  } = useForm<RecoverUserFormData>();
+
+  const initRecovery = async (data: InitRecoveryFormData) => {
+    const response = await axios.post("/api/initRecovery", {
+      email: data.email,
+      targetPublicKey: data.targetPublicKey,
+    });
+    setInitRecoveryResponse(response.data);
+  };
+
+  const recoverUser = async (data: RecoverUserFormData) => {
+    if (iframeStamper === null) {
+      throw new Error("iframeStamper is null");
+    }
+    if (initRecoveryResponse === null) {
+      throw new Error("initRecoveryResponse is null");
+    }
+
+    let injected = await iframeStamper.injectRecoveryBundle(
+      data.recoveryBundle
+    );
+    if (injected !== true) {
+      throw new Error("unexpected error while injecting recovery bundle");
+    }
+
+    const challenge = generateRandomBuffer();
+    const authenticatorUserId = generateRandomBuffer();
+
+    // An example of possible options can be found here:
+    // https://www.w3.org/TR/webauthn-2/#sctn-sample-registration
+    const attestation = await getWebAuthnAttestation({
+      publicKey: {
+        authenticatorSelection: {
+          residentKey: "preferred",
+          requireResidentKey: false,
+          userVerification: "preferred",
+        },
+        rp: {
+          id: "localhost",
+          name: "Turnkey Federated Passkey Demo",
+        },
+        challenge,
+        pubKeyCredParams: [
+          { type: publicKey, alg: es256 },
+          { type: publicKey, alg: rs256 },
+        ],
+        user: {
+          id: authenticatorUserId,
+          name: data.authenticatorName,
+          displayName: data.authenticatorName,
+        },
+      },
+    });
+
+    const client = new TurnkeyClient(
+      {
+        baseUrl: process.env.NEXT_PUBLIC_BASE_URL!,
+      },
+      iframeStamper
+    );
+
+    // TODO: switch this to be `client.recoverUser`!
```

Comment:
should be able to switch this now for staging :)

Reply:
Not quite since we don't have the updated SDK released yet. I'll see what I can do to pull an early version in. Otherwise we should be able to do this sometime this week when the proper release goes out.

Reply:
Done!